src/components/orator/js/Orator.js

```diff
@@ -0,0 +1,571 @@
+            play: {
+                changePath: "playing",
+                value: true,
+                source: "play"
+            },
+            pause: {
+                changePath: "playing",
+                value: false,
+                source: "pause"
+            },
+            toggle: {
+                funcName: "fluid.orator.controller.toggleState",
+                // Not providing an option for the explicit state value because
+                // when called through the jQuery click event the event object
+                // is passed in.
+                args: ["{that}", "playing"]
+            }
+        },
+        listeners: {
+            "onCreate.bindClick": {
+                "this": "{that}.dom.playToggle",
+                method: "click",
+                args: ["{that}.toggle"]
+            }
+        },
+        modelListeners: {
+            "playing": {
+                listener: "fluid.orator.controller.setToggleView",
+                args: ["{that}", "{change}.value"]
+            }
+        }
+    });
+
+    fluid.orator.controller.toggleState = function (that, path, state) {
+        var newState = fluid.isValue(state) ? state : !fluid.get(that.model, path);
+        // the !! ensures that the newState is a boolean value.
+        that.applier.change(path, !!newState, "ADD", "toggleState");
+    };
+
+    fluid.orator.controller.setToggleView = function (that, state) {
+        var playToggle = that.locate("playToggle");
+        playToggle.toggleClass(that.options.styles.play, state);
+        playToggle.attr({
+            "aria-label": that.options.strings[state ? "pause" : "play"]
+        });
+    };
+
+    fluid.orator.controller.container = function (containerSpec, defaultContainer, scope) {
+        var container = fluid.container(containerSpec, true);
+
+        if (container) {
+            return container;
+        }
+        var newContainer = $(defaultContainer);
+
+        // Unwrap to ensure that any jQuery element passed in actually has an element
+        // and if it does, we only want to use the first one.
+        scope = $(scope)[0] || "body";
+        $(scope).prepend(newContainer);
+        return fluid.container(newContainer);
+    };
+
+
+    /*******************************************************************************
+     * fluid.orator.domReader
+     *
+     * Reads in text from a DOM element and voices it
+     *******************************************************************************/
+
+    fluid.defaults("fluid.orator.domReader", {
+        gradeNames: ["fluid.viewComponent"],
+        selectors: {
+            highlight: ".flc-orator-highlight"
+        },
+        markup: {
+            highlight: "<mark class=\"flc-orator-highlight fl-orator-highlight\"></mark>"
+        },
+        events: {
+            onReadFromDOM: null
+        },
+        model: "{tts}.model",
+        members: {
+            parseQueue: [],
+            parseIndex: 0,
```

Comment:
I think a reasonable compromise would be to modelise i) parseIndex, and ii) parseQueue.length and key most of this component's activities off that - even if the current ChangeApplier doesn't make it a great idea to stash the entire queue in the model.